src/org/exist/protocolhandler/embedded/EmbeddedInputStream.java

```diff
@@ -33,66 +33,66 @@
 
 import org.exist.protocolhandler.xmldb.XmldbURL;
 import org.exist.storage.BrokerPool;
-import org.exist.storage.io.BlockingInputStream;
-import org.exist.storage.io.BlockingOutputStream;
 
 /**
- * Read document from embedded database as a (input)stream.
+ * Read document from embedded database as an InputStream
  *
  * @author Dannes Wessels
  */
 public class EmbeddedInputStream extends InputStream {
     
-    private final static Logger logger = LogManager.getLogger(EmbeddedInputStream.class);
+    private static final Logger LOG = LogManager.getLogger(EmbeddedInputStream.class);
     
-    private PipedInputStream  bis;
-    private PipedOutputStream bos;
-    private EmbeddedDownloadThread rt;
+    private final PipedInputStream  bis;
+    private final PipedOutputStream bos;
+    private final EmbeddedDownloadThread rt;
     
     /**
      *  Constructor of EmbeddedInputStream. 
      * 
      * @param xmldbURL Location of document in database.
      * @throws MalformedURLException Thrown for illegalillegal URLs.
      */
-    public EmbeddedInputStream(XmldbURL xmldbURL) throws IOException {
-        
+    public EmbeddedInputStream(final XmldbURL xmldbURL) throws IOException {
         this(null, xmldbURL);
     }
 
-        /**
+    /**
      *  Constructor of EmbeddedInputStream.
      *
      * @param xmldbURL Location of document in database.
      * @throws MalformedURLException Thrown for illegalillegal URLs.
      */
-    public EmbeddedInputStream(BrokerPool brokerPool, XmldbURL xmldbURL) throws IOException {
+    public EmbeddedInputStream(final BrokerPool brokerPool, final XmldbURL xmldbURL) throws IOException {
 
-        logger.debug("Initializing EmbeddedInputStream");
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("Initializing EmbeddedInputStream");
+        }
 
-        bis = new PipedInputStream(2048);
-        bos = new PipedOutputStream(bis);
+        this.bis = new PipedInputStream(4096);
```

Comment:
Cool. So to add further explanation to why that intention is important to me. As soon as I see mutation of shared state, I ask myself two questions:

1) Does this really need to be shared state? If not, I refactor to avoid it. Eliminating shared state allows me to ignore any worries of concurrent modification. Even if the code is currently intended to be single-threaded, the removal of shared state gives me the option to use this code concurrently in future.

2) If this does need to be shared state, what is the concurrency model? If it is anything other than single-threaded, what should I do to ensure the correct semantics when reading/writing this state.